### PR TITLE
[IA-1873] Build Go code into Leo image 

### PR DESCRIPTION
**Pull request by kyuksel**
[JIRA ticket
](https://broadworkbench.atlassian.net/browse/IA-1873)
[Companion firecloud-develop PR](https://github.com/broadinstitute/firecloud-develop/pull/2236)

Verified that the Leo container on FIAB has the shared lib and JAVA_OPTS includes `-Djna`.

```
root@fiab-kaan-real-adder:~# docker exec -it firecloud_leonardo-app_1 bash

bash-4.2# ls /leonardo/helm-go-lib/
go.mod	go.sum	libhelm.dylib  libhelm.h  main.go  README.md

bash-4.2# echo $JAVA_OPTS 
-server -Xms2g -Xmx2g ....... -Dsun.net.spi.nameservice.provider.2=dns,Jupyter -Djna.library.path=/leonardo/helm-go-lib
```

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -7,6 +20,7 @@ ENV GIT_HASH $GIT_HASH
 
 RUN mkdir /leonardo
 COPY ./leonardo*.jar /leonardo
+COPY --from=helm-go-lib-builder /helm-go-lib-build/helm-scala-sdk/helm-go-lib /leonardo/helm-go-lib
```
Comment (Qi77Qi) [inline Dockerfile:23]: shall we copy just the shared lib?

Comment (kyuksel) [reply Dockerfile:23]: We could but the other files are minuscule compared to `libhelm.dylib`. Not sure if worth the removal. Was also thinking `go.{mod,sum}` files could be useful for reference and checksumming?

**Review by Qi77Qi — APPROVED** [review]

**Review by rtitle — APPROVED** [review]
LGTM

**Comment by kyuksel** [comment]
jenkins retest

**Comment by kyuksel** [comment]
jenkins retest

**Comment by kyuksel** [comment]
@Qi77Qi @rtitle 2 Jenkins runs failed due to Sam and Rawls, which I can't imagine would be affected by my PR, can you?

**Comment by kyuksel** [comment]
jenkins retest

**Pull request merged**